src/include/sof/debug/debug.h

```diff
@@ -138,6 +138,7 @@
 #define dump_object_ptr(__o) do {} while (0)
 #endif
 
+#if !defined(__ZEPHYR__) || !CONFIG_LIBRARY
```

Comment:
probably do the same as arch/sof.h - lets see if this is easy to move to zephyr/include